components/todoist/actions/search-tasks/search-tasks.mjs

```diff
@@ -0,0 +1,64 @@
+import todoist from "../../todoist.app.mjs";
+
+export default {
+  key: "todoist-search-tasks",
+  name: "Search Tasks",
+  description: "Search tasks by name, label, project and/or section. [See Docs](https://developer.todoist.com/rest/v2/#get-active-tasks)",
+  version: "0.0.1",
+  type: "action",
+  props: {
+    todoist,
+    name: {
+      type: "string",
+      label: "Name Match",
+      description: "Returns tasks that contain the specified string in their name.",
+      optional: true,
+      default: "",
+    },
+    label: {
+      propDefinition: [
+        todoist,
+        "label",
+      ],
+    },
+    project: {
+      propDefinition: [
+        todoist,
+        "project",
+      ],
+    },
+    section: {
+      propDefinition: [
+        todoist,
+        "section",
+        (c) => ({
+          project: c.project,
+        }),
+      ],
+    },
+  },
+  async run ({ $ }) {
+    const {
+      name,
+      label,
+      project,
+      section,
+    } = this;
+    const tasks = await this.todoist.getActiveTasks({
+      $,
+      params: {
+        label: label,
+        project_id: project,
+        section_id: section,
+      },
+    });
+    let result = tasks.filter((task) => (task.content).includes(name));
```

Comment:
I believe you meant this instead:
```suggestion
    let result = tasks.filter((task) => (task.content.includes(name)));
```

Also, here you could check if `name` actually received a value from the user - else, no need to filter at all:
```suggestion
    let result = name ? tasks.filter((task) => (task.content.includes(name))) : tasks;
```